cloudinit/config/schema.py

```diff
@@ -173,17 +176,32 @@ def annotated_cloudconfig_file(cloudconfig, original_content, schema_errors):
 def validate_cloudconfig_file(config_path, schema, annotate=False):
     """Validate cloudconfig file adheres to a specific jsonschema.
 
-    @param config_path: Path to the yaml cloud-config file to parse.
+    @param config_path: Path to the yaml cloud-config file to pars, or
+        SYSTEM_USERDATA_FILE to indicate that the system userdata should be
+        queried.
     @param schema: Dict describing a valid jsonschema to validate against.
     @param annotate: Boolean set True to print original config file with error
         annotations on the offending lines.
 
     @raises SchemaValidationError containing any of schema_errors encountered.
     @raises RuntimeError when config_path does not exist.
     """
-    if not os.path.exists(config_path):
-        raise RuntimeError('Configfile {0} does not exist'.format(config_path))
-    content = load_file(config_path, decode=False)
+    if config_path == SYSTEM_USERDATA_FILE:
+        (content, err) = subp(
+            ["cloud-init", "query", "userdata"], decode=False
+        )
```

Comment:
@OddBloke The reason we treat this specially is because the typical userdata path in some cases is actually a compressed binary (juju deployed units). In this case the query command uses the special [cloudinit.cmd.query.load_userdata func to decompress and decode](https://github.com/canonical/cloud-init/blob/master/cloudinit/cmd/query.py#L81-L93) that I wanted to leverage here. It would probably be preferable to either call the function directly or relocate that utility function  into util.py.